lively.ide/components/reconciliation.js

```diff
@@ -808,10 +808,13 @@ class MorphRemovalReconciliation extends Reconciliation {
  */
 class MorphIntroductionReconciliation extends Reconciliation {
   reconcile () {
-    const { descriptor } = this;
-
-    const safeName = descriptor.ensureNoNameCollisionInDerived(this.addedMorph.name, true);
-    if (safeName !== this.addedMorph.name) this.addedMorph.name = safeName;
+    const { descriptor, addedMorph } = this;
+    const safeName = descriptor.ensureNoNameCollisionInDerived(addedMorph.name);
+    if (safeName !== this.addedMorph.name) {
+      addedMorph.withMetaDo({ reconcileChanges: false }, () => {
+        addedMorph.name = safeName; // do not reconcile this
+      });
```

Comment:
Can you explain this to me later on? But I can confirm that both bugs are gone, really nice!